src/measure_theory/interval_integral.lean

```diff
@@ -526,6 +526,95 @@ lemma integral_pos_iff_support_of_nonneg_ae {f : ℝ → ℝ} {a b : ℝ}
   0 < ∫ x in a..b, f x ↔ a < b ∧ 0 < volume (function.support f ∩ Ioc a b) :=
 integral_pos_iff_support_of_nonneg_ae' (ae_mono measure.restrict_le_self hf) hfi
 
+section mono
+
+variables {μ : measure ℝ} {f g : ℝ → ℝ} {a b : ℝ}
+  (hf : interval_integrable f μ a b) (hg : interval_integrable g μ a b)
+  (hab : a ≤ b)
+
+include hf hg hab
+
+lemma integral_mono_ae_restrict (h : f ≤ᵐ[μ.restrict (interval a b)] g) :
+∫ u in a..b, f u ∂μ ≤ ∫ u in a..b, g u ∂μ :=
+begin
+  rw [integral_of_le hab, integral_of_le hab],
+  rw interval_of_le hab at h,
+  exact set_integral_mono_ae_restrict hf.1 hg.1
+    (h.filter_mono (ae_mono $ measure.restrict_mono Ioc_subset_Icc_self (le_refl μ)))
+end
+
+lemma interval_integral_mono_ae (h : f ≤ᵐ[μ] g) :
+∫ u in a..b, f u ∂μ ≤ ∫ u in a..b, g u ∂μ :=
+begin
+  rw [integral_of_le hab, integral_of_le hab],
+  exact set_integral_mono_ae hf.1 hg.1 h
+end
+
+lemma interval_integral_mono_on (h : ∀ x ∈ interval a b, f x ≤ g x) :
```

Comment:
Is that really useful for an hypothesis ? I can see why you would want that for a definition or the conclusion of a lemma, because it simplifies *applying*. But to *prove* an hypothesis like that you would have to put `(λ x hx, ...)` or the tactic-mode equivalent in both cases, right ?

Reply:
I agree with @ADedecker, my understanding is that implicitness in hypotheses is only relevent within the proof itself, not to the user of the lemma.

Reply:
Sorry, you're right, I was thinking of a different issue.